lib/src/git.rs

```diff
@@ -136,14 +170,27 @@ pub fn import_some_refs(
             }
         };
         let id = CommitId::from_bytes(git_commit.id().as_bytes());
-        new_git_heads.insert(id.clone());
+        new_git_heads.insert(id.clone()); // TODO/unrelated: why is this before checking for git_ref_filter?
         if !git_ref_filter(&full_name) {
             continue;
         }
         // TODO: Make it configurable which remotes are publishing and update public
         // heads here.
         let old_target = existing_git_refs.remove(&full_name);
         let new_target = Some(RefTarget::Normal(id.clone()));
+        if let Some(RefName::LocalBranch(branch_name)) = parse_git_ref(&full_name) {
+            // Record the location of the branch in git repo into our view. Note that the
+            // branch may still become conflicted in jj below. Regardless we need to record
+            // where it was in git when we last synchronized the jj and git views, and the
+            // branch cannot be conflicted on the git side.
+            new_git_view.set_branch(
+                branch_name,
+                BranchTarget {
+                    local_target: Some(RefTarget::Normal(id.clone())),
+                    remote_targets: Default::default(),
+                },
+            );
```

Comment:
Do we also need to do this in the block on lines 204-205 (i.e. for deleted branches)?

Reply:
This is a good point. I need to write a test for this (or is there one already)?

We don't have a good UI for move-deletion branch conflicts. I'm not sure what should happen. The simplest thing to imagine would be to just recreate the branch as unconflicted if it was deleted in git and moved in JJ. WDYT, @martinvonz ?

Reply:
If we just delete the branch from "last seen refs view" on a delete-move conflict and do nothing else, it will become conflicted in jj and will no longer be exported by `jj git export`. I think the user might be confused by a conflicted branch that is in only one place and be surprised that it's not exported. The solution would be to either `jj branch set` or `jj branch delete` it.

Reply:
Good question. It sounds like the right thing to do is to leave it as conflict in the model and (later) improve the UX for it, or what do you think?

Reply:
Yes, probably. I might just follow your original suggestion and leave a TODO to actually have a test that can tell the difference and to think about UI.

The best idea for a UI I have so far would be to print a warning and make the branch un-conflicted. Printing a warning might be tricky, though.